scripts/source-maintenance.sh

```diff
@@ -351,21 +343,12 @@ for FILENAME in `git ls-files`; do
 	# Code Style formatting maintenance
 	#
 	applyPluginsTo ${FILENAME} scripts/maintenance/ || return
-	if test "${ASVER}"; then
-		./scripts/formater.pl --with-astyle ${ASTYLE} ${FILENAME}
-		if test -e $FILENAME -a -e "$FILENAME.astylebak"; then
-			md51=`cat  $FILENAME| tr -d "\n \t\r" | $MD5`;
-			md52=`cat  $FILENAME.astylebak| tr -d "\n \t\r" | $MD5`;
-
-			if test "$md51" != "$md52"; then
-				echo "ERROR: File $FILENAME not formatting well";
-				mv $FILENAME $FILENAME.astylebad
-				mv $FILENAME.astylebak $FILENAME
-				git checkout -- ${FILENAME}
-			else
-				rm -f $FILENAME.astylebak
-			fi
-        	fi
+	if test "$CppFormatter"; then
+		if $CppFormatter $FILENAME > $FILENAME.new; then
+			updateIfChanged $FILENAME $FILENAME.new 'by astyle'
+		else
+			rm $FILENAME.new
+		fi
```

Comment:
This is another place where we are automatically "recovering" from formatting errors. As discussed in another commit, I do not think we should be doing that, and I can polish this code further to fail on formatting problems. The developer running this script can always use `--keep-going` to get past early failures.